pkg/sitewise/auth.go

```diff
@@ -0,0 +1,107 @@
+package sitewise
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend/log"
+	"github.com/grafana/iot-sitewise-datasource/pkg/models"
+)
+
+type EdgeAuthenticator struct {
+	Settings models.AWSSiteWiseDataSourceSetting
+}
+
+func (a *EdgeAuthenticator) Authorize() (models.AuthInfo, error) {
```

Comment:
I think better to call everything `Authenticator`, `Authenticate` or `Authorizer`, `Authorize`